rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1578,19 +1578,32 @@ public void addSignature(BtcECKey federatorBtcPublicKey, List<byte[]> signatures
 
         Context.propagate(btcContext);
 
+        if (svpIsOngoing()) {
+            provider.getSvpSpendTxWaitingForSignatures()
+                .filter(svpSpendTxWFS -> svpSpendTxWFS.getKey().equals(rskTxHash))
+                .ifPresent(svpSpendTxWFS -> {
+                    logger.trace("Going to add federator public key {} to svp spend transaction", federatorBtcPublicKey);
+                    BtcTransaction svpSpendTx = svpSpendTxWFS.getValue();
+                    if (!areSignaturesEnoughToSignAllTxInputs(svpSpendTx, signatures)) {
+                        return;
+                    }
+                    addSvpSpendTxSignatures(federatorBtcPublicKey, signatures, rskTxHash, svpSpendTx);
+                });
+        }
+
         BtcTransaction releaseTx = provider.getPegoutsWaitingForSignatures().get(rskTxHash);
         if (releaseTx == null) {
             logger.warn("No tx waiting for signature for hash {}. Probably fully signed already.", rskTxHash);
             return;
         }
-        if (!hasEnoughSignatures(releaseTx, signatures)) {
+        if (!areSignaturesEnoughToSignAllTxInputs(releaseTx, signatures)) {
             return;
         }
 
         addReleaseSignatures(federatorBtcPublicKey, signatures, rskTxHash, releaseTx);
     }
 
-    private boolean hasEnoughSignatures(BtcTransaction releaseTx, List<byte[]> signatures) {
+    private boolean areSignaturesEnoughToSignAllTxInputs(BtcTransaction releaseTx, List<byte[]> signatures) {
```

Comment:
this method name was change to be distinguishable from BridgeUtils method thats also being used here

Reply:
Honestly, I don't like that name 😅. I prefer the old one or something like: `isBtcTranactionFullySigned`. Another thing that can help is moving this method to `BitcoinUtils`